flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -483,27 +485,42 @@ protected StreamTask(
             }
 
             this.systemTimerService = createTimerService("System Time Trigger for " + getName());
-
+            final CheckpointStorageAccess finalCheckpointStorageAccess = checkpointStorageAccess;
+
+            ChannelStateWriter channelStateWriter =
+                    configuration.isUnalignedCheckpointsEnabled()
+                            ? openChannelStateWriter(
+                                    getName(),
+                                    () -> {
+                                        if (finalCheckpointStorageAccess
+                                                instanceof FsMergingCheckpointStorageAccess) {
+                                            return finalCheckpointStorageAccess;
+                                        } else {
+                                            return checkpointStorage.createCheckpointStorage(
```

Comment:
Would you mind adding some comments here to explain why we don't pass `CheckpointStorageAccess` directly?